src/sqlitestorage.cpp

```diff
@@ -656,7 +656,7 @@ int SqliteStorage::Private::loadIncidencesBySeries(sqlite3_stmt *stmt1, QStringL
 
     while ((incidence = mFormat->selectComponents(stmt1, notebookUid))
            && (limit <= 0 || count < limit)) {
-        if (addIncidence(incidence, notebookUid)) {
+        if (addIncidence(Incidence::Ptr(incidence), notebookUid)) {
```

Comment:
This looks like potentially leaking memory. The selectComponents() returns an instance owned now by the caller, but then there are other filters above which can make this ctor skipped.

Perhaps safer if either the "incidence" instance is kept as Ptr or switched into some other similar type e.g. QScopedPointer, and the while loop assignment constructs a proper wrapped instance, or then we don't touch the return types at all if it should be Ptr always.

Maybe just better if we don't return instances owned by the caller?

Reply:
Ah, that's a good remark. Returning a pointer like that and not taking ownership immediately can be very error prone, as you demonstrated.

I'm withdrawing the third commit completely introducing this. Read routines will continue to return a QSharedPointer for the moment, up to when a real case for change appears.